addons/mail/static/src/models/thread_viewer/thread_viewer.js

```diff
@@ -118,7 +119,12 @@ function factory(dependencies) {
             const lastMessageIsVisible = this.lastVisibleMessage &&
                 this.lastVisibleMessage === this.lastMessage;
             if (lastMessageIsVisible && this.hasComposerFocus && this.thread) {
-                this.thread.markAsSeen();
+                this.thread.markAsSeen().catch(e => {
+                    // prevent crash when executing compute during destroy
+                    if (!(e instanceof RecordDeletedError)) {
+                        throw e;
+                    }
+                });
```

Comment:
I tried to prevent compute of destroyed record but it didn't go well. The big issue here is that the compute is triggering an async method, and at the time of the compute the record is not yet marked as deleted.